plugins/dkim_sign.js

```diff
@@ -166,32 +166,44 @@ exports.load_key = function (file) {
 exports.hook_queue_outbound = exports.hook_pre_send_trans_email = function (next, connection) {
     const plugin = this;
     if (plugin.cfg.main.disabled) return next();
+
     if (connection.transaction.notes.dkim_signed) {
-        connection.logdebug(plugin, 'email already signed');
+        connection.logdebug(plugin, 'already signed');
         return next();
     }
 
-    plugin.get_key_dir(connection, function (err, keydir) {
+    let selector;
+    let private_key;
+    let domain = plugin.get_sender_domain(connection);
+
+    if (!domain) {
```

Comment:
@msimerson 

This broke [single domain configuration](https://haraka.github.io/manual/plugins/dkim_sign.html)

before this, the absence of per domain keydir switched you to single domain mode
```js
        if (!keydir) {
            domain = plugin.cfg.main.domain;
            private_key = plugin.private_key;
            selector = plugin.cfg.main.selector;
        }
```